__tests__/utils/cache.test.js

```diff
@@ -0,0 +1,31 @@
+import {act} from 'react-test-renderer';
+import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/async-storage-mock';
+
+import {getSpecificCache, removeAllCache, saveToCache} from '../../src/utils/cache';
+
+describe('utils cache should run correctly', () => {
+  it('remove all cache shoule run correctly', () => {
+    const mockRemoveCache = jest.spyOn(mockAsyncStorage, 'clear');
+    act(() => {
+      removeAllCache();
+    });
+    expect(mockRemoveCache).toHaveBeenCalled();
+  });
+  it('saveToCache should run correctly', () => {
+    const callback = jest.fn().mockImplementation(() => undefined, 'cool');
+
+    const mockgetItem = jest
+      .spyOn(mockAsyncStorage, 'getItem')
+      .mockImplementation(('cache', callback));
+    act(() => {
+      saveToCache('cache', {name: 'agita'});
+    });
+    expect(mockgetItem).toHaveBeenCalled();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The test checks if `mockgetItem` has been called, but the actual function being tested is `saveToCache`. You should check if `mocksetItem` has been called instead.
```diff
- expect(mockgetItem).toHaveBeenCalled();
+ const mocksetItem = jest.spyOn(mockAsyncStorage, 'setItem');
+ expect(mocksetItem).toHaveBeenCalled();
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->